services/ui-src/src/components/reports/ModalOverlayReportPage.tsx

```diff
@@ -13,153 +14,214 @@ import {
+        (entity: EntityShape) => entity.id === currentEntity?.id
+      );
+      const filteredFormData = filterFormData(
+        enteredData,
+        overlayForm!.fields.filter(isFieldElement)
+      );
+      const entriesToClear = getEntriesToClear(
+        enteredData,
+        overlayForm!.fields.filter(isFieldElement)
+      );
+      const newEntity = {
+        ...currentEntity,
+        ...filteredFormData,
+      };
+      let newEntities = currentEntities;
+      newEntities[selectedEntityIndex] = newEntity;
+      newEntities[selectedEntityIndex] = setClearedEntriesToDefaultValue(
+        newEntities[selectedEntityIndex],
+        entriesToClear
+      );
+      const shouldSave = entityWasUpdated(
+        reportFieldDataEntities[selectedEntityIndex],
+        newEntity
+      );
+      if (shouldSave) {
+        const dataToWrite = {
+          metadata: {
+            status: ReportStatus.IN_PROGRESS,
+            lastAlteredBy: full_name,
+          },
+          fieldData: {
+            [entityType]: newEntities,
+          },
+        };
+        await updateReport(reportKeys, dataToWrite);
+      }
+      setSubmitting(false);
+    }
+    closeEntityDetailsOverlay();
+    setSidebarHidden(false);
+  };
+
   return (
     <Box>
-      {overlayForm && selectedEntity && isEntityDetailsOpen ? (
+      {overlayForm && isEntityDetailsOpen && currentEntity ? (
         <EntityProvider>
           <EntityDetailsOverlay
+            closeEntityDetailsOverlay={closeEntityDetailsOverlay}
             entityType={entityType as EntityType}
-            selectedEntity={selectedEntity}
+            entities={report?.fieldData[entityType]}
             form={overlayForm}
-            verbiage={verbiage}
-            closeEntityDetailsOverlay={closeEntityDetailsOverlay}
-            setSidebarHidden={setSidebarHidden}
+            onSubmit={onSubmit}
+            selectedEntity={currentEntity}
+            disabled={!userIsEndUser}
+            submitting={submitting}
             validateOnRender={validateOnRender}
           />
         </EntityProvider>
       ) : (
         <Box sx={sx.content}>
-          {verbiage.intro && (
-            <ReportPageIntro
-              text={verbiage.intro}
-              accordion={
-                accordionVerbiage[reportType as keyof typeof accordionVerbiage]
-                  ?.formIntro
-              }
-              reportType={reportType}
-            />
-          )}
+          <ReportPageIntro
+            text={verbiage.intro}
+            accordion={accordionVerbiage.MLR.formIntro}
+            reportType={report?.reportType}
+          />
+
           <Box sx={sx.dashboardBox}>
             <Heading as="h3" sx={sx.dashboardTitle}>
               {dashTitle}
             </Heading>
             {reportFieldDataEntities.length === 0 ? (
               <>
-                <hr />
+                <Box sx={sx.tableSeparator} />
```

Comment:
The separator was the wrong color and didn't have enough spacing to match the mockups